File: src/dataset/typescript/luhn_Exercism-typescript-thewanionly/claude/ClaudeSonnet4_luhn_v1.ts
export function valid(digitString: string): boolean {
  // Remove whitespaces and check length in one pass
  const cleanString = digitString.replace(/\s/g, '');
  
  // Early return for invalid length
  if (cleanString.length <= 1) return false;
  
  let sum = 0;
  let isEven = false;
  
  // Process digits from right to left in a single pass
  for (let i = cleanString.length - 1; i >= 0; i--) {
    const char = cleanString[i];
    
    // Check if character is a digit
    if (char < '0' || char > '9') return false;
    
    let digit = parseInt(char, 10);
    
    // Double every second digit from the right
    if (isEven) {
      digit *= 2;
      if (digit > 9) digit -= 9;
    }
    
    sum += digit;
    isEven = !isEven;
  }
  
  return sum % 10 === 0;
}

File: src/dataset/go/luhn_Exercism_go_drapala/claude/ClaudeSonnet4_luhn_v1.go
package luhn

func Valid(number string) bool {
	// Remove spaces and validate in one pass
	clean := make([]byte, 0, len(number))
	for i := 0; i < len(number); i++ {
		c := number[i]
		if c == ' ' {
			continue
		}
		if c < '0' || c > '9' {
			return false
		}
		clean = append(clean, c)
	}

	// Check length
	if len(clean) <= 1 {
		return false
	}

	sum := 0
	isEven := false

	// Process from right to left
	for i := len(clean) - 1; i >= 0; i-- {
		digit := int(clean[i] - '0')
		
		if isEven {
			digit *= 2
			if digit > 9 {
				digit -= 9
			}
		}
		
		sum += digit
		isEven = !isEven
	}

	return sum%10 == 0
}